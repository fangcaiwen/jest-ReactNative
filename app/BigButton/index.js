/**
 * Created by 208439 on 2019/4/3
 *
 * Author: wind
 *
 * Content:点击的button
 */

import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    TouchableOpacity,
} from 'react-native';

export default class BigButton extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { title,callBack } = this.props;
        return (
            <TouchableOpacity style={styles.container} onPress={() => callBack()}>
                <View style={{width:300,height:50,alignItems: 'center',justifyContent:'center',backgroundColor:'#00b7e3',borderRadius:4}}>
                    <Text style={{color:'#fff',fontSize:18}}>{title}</Text>
                </View>
            </TouchableOpacity>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        width:300,
        height:50
    },
});