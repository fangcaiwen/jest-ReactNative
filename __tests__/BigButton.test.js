/**
 * Created by 208439 on 2019/4/3
 *
 * Author: wind
 *
 * Content:button测试用例
 */
import React from 'react';
import {configure, shallow,mount} from 'enzyme';
import BigButton from '../app/BigButton/index';
import renderer from "react-test-renderer";

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Testing Bigbutton component', () => {

    // 测试UI
    it('renders as expected', () => {
        const tree = renderer.create(<BigButton />).toJSON();
        // 快照测试
        expect(tree).toMatchSnapshot();
    });

    // 测试button传值以及点击事件
    it('renders BigButton as logic expected', () => {
        const wrapper = shallow(
            <BigButton title={'提交'} callBack={jest.fn(() => true)} />
        );
        // wrapper.props();
        // 测试点击功能
        expect(wrapper.props().onPress()).toBeTruthy();

        // 测试title渲染功能
        expect(wrapper.props().children.props.children.props.children).toBe('提交');
    });

});
