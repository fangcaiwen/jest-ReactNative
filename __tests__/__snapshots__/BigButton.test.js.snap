// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Bigbutton component renders as expected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "opacity": 1,
      "width": 300,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00b7e3",
        "borderRadius": 4,
        "height": 50,
        "justifyContent": "center",
        "width": 300,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;
